packages/systemd/9009-machine-id-setup-generate-stable-ID-under-VMWare.patch

```diff
@@ -0,0 +1,66 @@
+From f18ea74a0a62436849cf26ba9b8d5ab78d4f61a5 Mon Sep 17 00:00:00 2001
+From: Arnaldo Garcia Rincon <agarrcia@amazon.com>
+Date: Wed, 28 Apr 2021 22:23:23 +0000
+Subject: [PATCH 9009/9009] machine-id-setup: generate stable ID under VMWare
+
+Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>
+---
+ src/core/machine-id-setup.c | 22 ++++++++++++++++++----
+ 1 file changed, 18 insertions(+), 4 deletions(-)
+
+diff --git a/src/core/machine-id-setup.c b/src/core/machine-id-setup.c
+index aa9609f..7c497a2 100644
+--- a/src/core/machine-id-setup.c
++++ b/src/core/machine-id-setup.c
+@@ -31,6 +31,8 @@ static int generate_machine_id(const char *root, sd_id128_t *ret) {
+         const char *dbus_machine_id;
+         _cleanup_close_ int fd = -1;
+         int r;
++        int virtualization_type;
++
+ 
+         assert(ret);
+ 
+@@ -60,7 +62,11 @@ static int generate_machine_id(const char *root, sd_id128_t *ret) {
+                                 return 0;
+                         }
+ 
+-                } else if (detect_vm() == VIRTUALIZATION_KVM) {
++                }
++
++                virtualization_type = detect_vm();
++
++                if (virtualization_type == VIRTUALIZATION_KVM) {
+ 
+                         /* If we are not running in a container, see if we are
+                          * running in qemu/kvm and a machine ID was passed in
+@@ -76,15 +82,23 @@ static int generate_machine_id(const char *root, sd_id128_t *ret) {
+                                 return 0;
+                         }
+ 
+-                } else if (detect_vm() == VIRTUALIZATION_XEN) {
++                } else if (virtualization_type == VIRTUALIZATION_XEN) {
+ 
+-                        /* Otherwise, if we're running under Xen, check for
+-                         * a UUID from the hypervisor. */
++                        /* If we're running under Xen, check for a UUID from
++                         * the hypervisor. */
+ 
+                         if (id128_read("/sys/hypervisor/uuid", ID128_UUID, ret) >= 0) {
+                                 log_info("Initializing machine ID from Xen UUID.");
+                                 return 0;
+                         }
++                } else if (virtualization_type == VIRTUALIZATION_VMWARE) {
++                        /* If we're running under VMWare, check for a machine ID
++                         * passed in via --uuid */
```

Comment:
I don't think this comment is accurate for VMWare.

Reply:
Also - the company styles it as `VMware`, let's follow suit. :)